src/tables.jl

```diff
@@ -207,16 +239,166 @@ function _colnames(s::AbstractDimStack)
+        return WideDimTable(layers..., layernames=layernames, mergedims=mergedims)
+    else
+        # Construct DimColumns
+        dims_ = dims(x)
+        dimcolumns = map(d -> DimColumn(d, dims_), dims_)
+        dimnames = collect(map(dim2key, dims_))
+
+        # Construct DimArrayColumn
+        dimarraycolumn = DimArrayColumn(x, dims_)
+
+        # Merge DimColumns
+        if mergedims
+            colnames = vcat([:geometry], [:value])
+            dimcol = MergedDimColumn(Tuple(dimcolumns), :geometry)
+            return WideDimTable{typeof(dimcol)}(colnames, dimcol, [dimarraycolumn])
+        else
+            return WideDimTable{typeof(dimcolumns)}(vcat(dimnames, [:value]), dimcolumns, [dimarraycolumn])
+        end
+    end
+end
+
+dimcolumns(t::WideDimTable) = getfield(t, :dimcolumns)
+dimarraycolumns(t::WideDimTable) = getfield(t, :dimarraycolumns)
+dims(t::WideDimTable) = dims(parent(t))
+
+Base.parent(t::WideDimTable) = getfield(t, :colnames)
+
+for func in (:dims, :val, :index, :lookup, :metadata, :order, :sampling, :span, :bounds,
+             :locus, :name, :label, :units)
+    @eval $func(t::WideDimTable, args...) = $func(parent(t), args...)
+
+end
+
+Tables.istable(::WideDimTable) = true
+Tables.columnaccess(::Type{<:WideDimTable}) = true
+Tables.columns(t::WideDimTable) = t
+Tables.columnnames(c::WideDimTable) = parent(c)
+
+function Tables.schema(t::WideDimTable) 
+    colnames = parent(t)
+    types = vcat([map(eltype, dimcolumns(t))...], [map(eltype, dimarraycolumns(t))...])
+    Tables.Schema(colnames, types)
+end
+
+function Tables.schema(t::WideDimTable{<:MergedDimColumn}) 
+    colnames = parent(t)
+    types = vcat([eltype(dimcolumns(t))], [map(eltype, dimarraycolumns(t))...])
+    Tables.Schema(colnames, types)
+end
+
+@inline function Tables.getcolumn(t::WideDimTable, key::Symbol)
+    keys = parent(t)
+    i = findfirst(==(key), keys)
+    n_dimcols = length(dimcolumns(t))
+    if i <= n_dimcols
+        return dimcolumns(t)[i]
+    else
+        return dimarraycolumns(t)[i - n_dimcols]
+    end
+end
+
+@inline function Tables.getcolumn(t::WideDimTable{<:MergedDimColumn}, key::Symbol)
+    keys = parent(t)
+    i = findfirst(==(key), keys)
+    if i == 1
+        return dimcolumns(t)
+    else
+        return dimarraycolumns(t)[i - 1]
+    end
+end
+
+@inline function Tables.getcolumn(t::WideDimTable, ::Type{T}, i::Int, key::Symbol) where T
+    Tables.getcolumn(t, key)
+end
+
+
 # TableTraits.jl interface
 
+
 function IteratorInterfaceExtensions.getiterator(x::DimTableSources)
-    return Tables.datavaluerows(Tables.columntable(x))
+    return Tables.datavaluerows(Tables.dictcolumntable(x))
 end
 IteratorInterfaceExtensions.isiterable(::DimTableSources) = true
 TableTraits.isiterabletable(::DimTableSources) = true
 
 function IteratorInterfaceExtensions.getiterator(t::DimTable)
-    return Tables.datavaluerows(Tables.columntable(t))
```

Comment:
What is the reasoning for changing these to `dictcolumntable` ?

Reply:
The main reason is that `columntable` produces a named tuple with one field for each layer/column, which is going to be slow when we have several hundred.

Reply:
Makes sense. But this is on DimTable rather than `WideDimTable`

Reply:
I've replace `WideDimTable` with `DimTable`.